src/component/DefaultLegendContent.tsx

```diff
@@ -23,7 +22,10 @@ export type VerticalAlignmentType = 'top' | 'bottom' | 'middle';
 export type Formatter = (value: any, entry: LegendPayload, index: number) => ReactNode;
 
 export interface LegendPayload {
-  value: any;
+  /**
+   * This is the text that will be displayed in the legend in the DOM.
+   */
+  value: string;
```

Comment:
Is the value always a string?

Reply:
I made it a string now. It used to always pass the `dataKey` through and that could be a function.